src/libraries/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -336,6 +342,7 @@ public sealed partial class RegexParseException : System.ArgumentException
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
         public override void GetObjectData(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
     }
+    [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
```

Comment:
Might be a dumb question, but does this only affect intellisense or could it also affect somehow the experience when looking at the source generated output? For instance, if someone is inspecting the source generated code for their expression, and want to F12 into the base types (RegexRunner or RegexRunnerFactory) would that still work and show the type definition?

Reply:
As an example, consider the SocketTaskExtensions type that's EditorBrowsableState.Never.  If I look in completion lists, it doesn't show up:
<img width="410" alt="image" src="https://user-images.githubusercontent.com/2642209/232903957-192af9ca-5e25-48b2-aeaa-01edf732b624.png">
nor do I get help from ctrl-space while typing to auto-complete:
<img width="289" alt="image" src="https://user-images.githubusercontent.com/2642209/232904073-29f91db4-fd9d-4433-b2b1-788fc844ded4.png">
but once I have typed it out, it still gets colorized, I can still dot off it:
<img width="473" alt="image" src="https://user-images.githubusercontent.com/2642209/232904285-d976603a-5e48-40b3-8200-a4137d0d0548.png">
I can still peek / go-to-definition:
<img width="658" alt="image" src="https://user-images.githubusercontent.com/2642209/232904396-ee767760-be30-4e16-88b5-a046944cca3d.png">
Etc.

This should not meaningfully negatively impact the source generator experience.